staging/src/k8s.io/kubectl/pkg/util/templates/templater.go

```diff
@@ -66,6 +66,29 @@ type templater struct {
 	Filtered []string
 }
 
+type FlagError struct {
```

Comment:
I'd replace all of this with just:
```go
func (templater *templater) FlagErrorFunc(exposedFlags ...string) func(*cobra.Command, error) error {
    return func(c *cobra.Command, err error) error {
        c.SilenceUsage = true
        return fmt.Errorf("%s\nSee '%s --help' for usage.", err, c.CommandPath())
    }
}
```

Reply:
:smiley: yup

Reply:
I've made the above change-it's much better, but with only this change something bugs me:
```console
$ kubectl options --foo
Error: unknown flag: --foo
See 'kubectl options --help' for usage.
```
but `kubectl options --help` returns nothing. 
I added a special case for the options --badflag to get:
```console
$ kubectl options --foo
Error: unknown flag: --foo
Run 'kubectl options' without flags.
```